mlflow/paddle/_paddle_autolog.py

```diff
@@ -0,0 +1,150 @@
+            self._log_metrics(logs, epoch)
+            self.epoch = epoch
+
+    def on_train_begin(self, logs=None):
+        self.client.set_tags(self.run_id, {"Mode": "training"})
+        params = {
+            "optimizer_name": _get_optimizer_name(self.model._optimizer),
+            "learning_rate": self.model._optimizer._learning_rate,
+        }
+        self.client.log_params(self.run_id, params)
+        self.client.flush(synchronous=True)
+
+    def on_train_end(self, logs=None):
+        self.metrics_logger.flush()
+        self.client.flush(synchronous=True)
+
+    def on_eval_end(self, logs=None):
+        eval_logs = {
+            "eval_" + key: (metric[0] if isinstance(metric, list) else metric)
+            for key, metric in logs.items()
+        }
+        self._log_metrics(eval_logs, self.epoch)
+
+
+def _log_early_stop_params(early_stop_callback, client, run_id):
+    """
+    Logs early stopping configuration parameters to MLflow.
+    :param early_stop_callback: The early stopping callback instance used during training.
+    :param client: An `MlflowAutologgingQueueingClient` instance used for MLflow logging.
+    :param run_id: The ID of the MLflow Run to which to log configuration parameters.
+    """
+    client.log_params(
+        run_id,
+        {
+            p: getattr(early_stop_callback, p)
+            for p in ["monitor", "patience", "min_delta", "stopped_epoch"]
+            if hasattr(early_stop_callback, p)
+        },
+    )
+
+
+def _log_early_stop_metrics(early_stop_callback, client, run_id):
+    """
+    Logs early stopping behavior results (e.g. stopped epoch) as metrics to MLflow.
+    :param early_stop_callback: The early stopping callback instance used during training.
+    :param client: An `MlflowAutologgingQueueingClient` instance used for MLflow logging.
+    :param run_id: The ID of the MLflow Run to which to log configuration parameters.
+    """
+    if early_stop_callback.stopped_epoch == 0:
+        return
+
+    metrics = {
+        "stopped_epoch": early_stop_callback.stopped_epoch,
+        "restored_epoch": early_stop_callback.stopped_epoch - max(1, early_stop_callback.patience),
+        "best_value": early_stop_callback.best_value,
+        "wait_epoch": early_stop_callback.wait_epoch,
+    }
+
+    client.log_metrics(run_id, metrics)
+
+
+def patched_fit(original, self, *args, **kwargs):
+    from paddle import callbacks
+
+    run_id = mlflow.active_run().info.run_id
+    tracking_uri = mlflow.get_tracking_uri()
+    client = MlflowAutologgingQueueingClient(tracking_uri)
+    metrics_logger = BatchMetricsLogger(run_id, tracking_uri)
+
+    log_models = get_autologging_config(mlflow.paddle.FLAVOR_NAME, "log_models", True)
+    log_every_n_epoch = get_autologging_config(mlflow.paddle.FLAVOR_NAME, "log_every_n_epoch", 1)
+
+    early_stop_callback = None
+    if "callbacks" in kwargs:
+        train_callbacks = kwargs["callbacks"]
```

Comment:
Can we rename this variable to `callbacks`? `train` sounds like there are `test` or `validation` callbacks, but there aren't.

Reply:
Sure!